### PR TITLE
Add README

### PR DESCRIPTION
**Pull request by marun**
Included is preliminary developer-focused documentation.

cc: @font @sdminonne @pmorie @onyiny-ang

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+$ apiserver-boot build generated
+```
+
+The generated code will need to be updated whenever the code for a
+type is modified. Care should be taken to separate generated from
+non-generated code in the commit history.
+
+### Enabling federation of a type
+
+Implementating support for federation of a Kubernetes type requires
```
Comment (sdminonne) [inline README.md]: Implemen~~ta~~ting

**Comment by sdminonne** [comment]
LGTM (nit: 1 typo)

**Comment by onyiny-ang** [comment]
LGTM 2

File: README.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+### Adding a new API type
+
+As per the
+[docs](https://github.com/kubernetes-incubator/apiserver-builder/blob/master/docs/tools_user_guide.md#create-an-api-resource)
+for apiserver-builder, bootstrapping a new fnord API type can be
+accomplished as follows:
+
+```
+# Bootstrap and commit a new type
+$ apiserver-boot create group version resource --group federation.k8s.io --version v1alpha1 --kind <your-kind>
```
Comment (sdminonne) [inline README.md]: --group federation~~.k8s.io~~ since `k8s.io` is added and `--group must match regex ^[a-z]+$ but was (federation.k8s.io)`

Comment (marun) [reply README.md]: Good catch!

**Comment by marun** [comment]
Thanks for the review!

**Pull request merged**